### PR TITLE
Access field admin

### PR DESCRIPTION
**Pull request by alexandraciobica**
### Summary of Changes

- Added access field to 
   - the user profile and 
   - the list of users in batch processing. 
- Added SQL and PostgreSQL in 
   - `administrator/components/com_admin/sql/updates/` and
   - `installation/sql/`

### CODE REVIEW

File: installation/sql/mysql/joomla.sql [hunk 1]
```diff
@@ -1994,6 +1994,7 @@ CREATE TABLE IF NOT EXISTS `#__users` (
   `otpKey` varchar(1000) NOT NULL DEFAULT '' COMMENT 'Two factor authentication encrypted keys',
   `otep` varchar(1000) NOT NULL DEFAULT '' COMMENT 'One time emergency passwords',
   `requireReset` tinyint(4) NOT NULL DEFAULT 0 COMMENT 'Require user to reset password on next login',
+  `access` int(10) unsigned NOT NULL DEFAULT 1,
```
Comment (laoneo) [inline installation/sql/mysql/joomla.sql]: We should disable access to the users initially. Like that we do not run into security issues. I would default it to 0.

File: administrator/components/com_users/Model/UsersModel.php [hunk 2]
```diff
@@ -96,6 +97,9 @@ protected function populateState($ordering = 'a.name', $direction = 'asc')
 
 		$this->setState('filter.excluded', $excluded);
 
+		$access = json_decode(base64_decode($app->input->get('access', '', 'BASE64')));
```
Comment (laoneo) [inline administrator/components/com_users/Model/UsersModel.php]: This looks like something is wrong here.

Comment (alexandraciobica) [reply administrator/components/com_users/Model/UsersModel.php]: I modified it according to how it is done in ArticlesModel, is it ok now?

File: administrator/components/com_users/Model/UsersModel.php [hunk 3]
```diff
@@ -50,6 +50,7 @@ public function __construct($config = array(), MVCFactoryInterface $factory = nu
 				'range',
 				'lastvisitrange',
 				'state',
+                'access', 'a.access', 'access_level',
```
Comment (laoneo) [inline administrator/components/com_users/Model/UsersModel.php]: Please use tabs

File: administrator/components/com_users/Model/UsersModel.php [hunk 4]
```diff
@@ -96,6 +97,9 @@ protected function populateState($ordering = 'a.name', $direction = 'asc')
 
 		$this->setState('filter.excluded', $excluded);
 
+		$access = $app->input->post->get('access');
```
Comment (laoneo) [inline administrator/components/com_users/Model/UsersModel.php]: You can just use `$app->input->getInt('access');`

File: administrator/components/com_users/Model/UsersModel.php [hunk 5]
```diff
@@ -50,6 +50,7 @@ public function __construct($config = array(), MVCFactoryInterface $factory = nu
 				'range',
 				'lastvisitrange',
 				'state',
+				'access', 'a.access', 'access_level',
```
Comment (bembelimen) [inline administrator/components/com_users/Model/UsersModel.php:53]: Should be line by line

File: administrator/components/com_users/Model/UsersModel.php [hunk 6]
```diff
@@ -125,6 +129,7 @@ protected function getStoreId($id = '')
 		$id .= ':' . $this->getState('filter.state');
 		$id .= ':' . $this->getState('filter.group_id');
 		$id .= ':' . $this->getState('filter.range');
+		$id .= ':' . serialize($this->getState('filter.access'));
```
Comment (bembelimen) [inline administrator/components/com_users/Model/UsersModel.php]: Why serialize, when it's only an INT? (line 101)

File: administrator/components/com_users/Model/UsersModel.php [hunk 7]
```diff
@@ -295,6 +300,10 @@ protected function getListQuery()
 			}
 		}
 
+		// Join over the asset groups.
+		$query->select('ag.title AS access_level')
+			->join('LEFT', '#__viewlevels AS ag ON ag.id = a.access');
```
Comment (bembelimen) [inline administrator/components/com_users/Model/UsersModel.php]: ```->leftJoin(...)```?

Comment (bembelimen) [reply administrator/components/com_users/Model/UsersModel.php]: Please always(!!) escape all fields
```$db->quoteName(...)```

File: administrator/components/com_users/Model/UsersModel.php (hunk 5 above)
Comment (alexandraciobica) [reply administrator/components/com_users/Model/UsersModel.php:53]: It is done like this in ArticlesModel, should I change it here?

**Pull request merged**